### PR TITLE
Resolve logic task bowling game and write unit test for it

### PR DESCRIPTION
**Pull request by ChebanTetyana**
Create classes with main game. This game will be bowling. Check all rules in internet. Should have one public function ‘roll’ that will:

check if player can roll ball
roll and randomly hit from 1 to 10 pin. Return number and update state of game.
If you hit strike or pin. It will return 10 and skip round correctly.
Second public method:

print current result as array.
Mimic game in tests.

### CODE REVIEW

File: src/BowlingGame/BowlingGame.php [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            }
+        }
+
+        if ($this->currentFrame > 10) {
+            return "Game over!";
+        }
+        return $pinsHit;
+    }
+
+    public function print() :int
```
Comment (timiTao) [inline src/BowlingGame/BowlingGame.php]: add another method that will print history of throws

File: Tests/BowlingGameTest/BowlingGameTest.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+{
+    public function testRollStrike()
+    {
+        $game = new BowlingGame();
+        $pinsHit = $game->roll(10);
+
+        $this->assertEquals(10, $pinsHit);
+    }
+
+    public function testRollSpare()
```
Comment (timiTao) [inline Tests/BowlingGameTest/BowlingGameTest.php]: write some extra test with TDD

File: Tests/BowlingGameTest/BowlingGameTest.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        $this->assertEquals("Game over!", $game->roll(1));
+    }
+
+    public function testFinalScore()
+    {
+        $game = new BowlingGame();
+        for ($i = 0; $i < 12; $i++) {
+            $game->roll(10);
+        }
+        $this->assertEquals(320, $game->print());
```
Comment (timiTao) [inline Tests/BowlingGameTest/BowlingGameTest.php]: that should be 300 max

File: Tests/BowlingGameTest/BowlingGameTest.php [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    public function testRollSpare()
+    {
+        $this->game->roll(7);
+        $secondRoll = $this->game->roll(3);
+
+        $this->assertEquals(3, $secondRoll);
+        $this->assertEquals(10, $this->game->getResult()[0]);
+    }
+
+    public function testRoll()
```
Comment (timiTao) [inline Tests/BowlingGameTest/BowlingGameTest.php]: add return types

File: Tests/BowlingGameTest/BowlingGameTest.php [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    public function testFinalScore()
+    {
+        for ($i = 0; $i < 12; $i++) {
+            $this->game->roll(10);
+        }
+        $scores = $this->game->getResult();
+        $this->assertEquals(300, end($scores));
+    }
+
+    public function testIncompleteGameScore(): void
```
Comment (timiTao) [inline Tests/BowlingGameTest/BowlingGameTest.php:105]: add test for 
* 3 last throws
* examples of full games with dataProvider

File: src/BowlingGame/BowlingGame.php [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+
+        return $frameScore;
+    }
+
+    public function isGameOver(): bool
+    {
+        return $this->currentFrame > 10;
+    }
+
+    public function validatePins(int $pinsHit): void
```
Comment (timiTao) [inline src/BowlingGame/BowlingGame.php]: this and others shouldn't be public

File: src/BowlingGame/BowlingGame.php (hunk 1 above)
Comment (ChebanTetyana) [reply src/BowlingGame/BowlingGame.php]: added method getResult

File: Tests/BowlingGameTest/BowlingGameTest.php (hunk 3 above)
Comment (ChebanTetyana) [reply Tests/BowlingGameTest/BowlingGameTest.php]: Rewrote to be 300 max

File: Tests/BowlingGameTest/BowlingGameTest.php (hunk 4 above)
Comment (ChebanTetyana) [reply Tests/BowlingGameTest/BowlingGameTest.php]: Added

File: Tests/BowlingGameTest/BowlingGameTest.php (hunk 5 above)
Comment (ChebanTetyana) [reply Tests/BowlingGameTest/BowlingGameTest.php:105]: Added testThreeFinalThrowsAfterStrike, testThreeFinalThrowsAfterSpare and testFullGame with dataProvider

File: src/BowlingGame/BowlingGame.php (hunk 6 above)
Comment (ChebanTetyana) [reply src/BowlingGame/BowlingGame.php]: Changed a few methods to private

File: Tests/BowlingGameTest/BowlingGameTest.php (hunk 4 above)
Comment (timiTao) [reply Tests/BowlingGameTest/BowlingGameTest.php]: still lacking

**Comment by timiTao** [comment]
Still need to fix 1 comment 👍

File: Tests/BowlingGameTest/BowlingGameTest.php (hunk 4 above)
Comment (ChebanTetyana) [reply Tests/BowlingGameTest/BowlingGameTest.php]: Added, thank you

**Review by timiTao — APPROVED** [review]

**Pull request merged**